lib/promscrape/discovery/ovhcloud/common.go

```diff
@@ -0,0 +1,105 @@
+package ovhcloud
+
+import (
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/netip"
+	"strconv"
+	"time"
+)
+
+func getAuthHeaders(cfg *apiConfig, headers http.Header, endpoint, path string) (http.Header, error) {
```

Comment:
It doesn't seem that any caller of this func checks for the error. Is this intended?

Reply:
This error should have been checked, and the SD HTTP request should be canceled.

However, it appears that `GetAPIResponseWithReqParams` does not consider the error from `RequestCallback func(req *http.Request)`. As a result, it sends out the service discovery request without the appropriate Auth Headers.

I am returning an error in `getAuthHeaders` and expecting it to be checked. Should we add a #TODO here or find a way to stop the SD request?

Reply:
Is "X-Ovh-Timestamp" mandatory?
If request to get server time fails, can we just use current time or `time()-configCheckInterval` as a value? This would simplify this function

Reply:
`X-Ovh-Timestamp` is mandatory for APIs that need auth.

I replaced `getTimeDelta` with `getOVHTimestamp`, which returns the server timestamp directly without `err`. It uses `now()` as a fallback and preserves the possibility to fetch the server timestamp in the next SD cycle.